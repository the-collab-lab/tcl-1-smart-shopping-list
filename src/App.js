import React from "react";
import AddItem from "./components/AddItem";
import ItemList from "./components/ItemList";

function App() {
  return (
    <div>
      <AddItem />
      <ItemList />
    </div>
  );
}

export default App;
