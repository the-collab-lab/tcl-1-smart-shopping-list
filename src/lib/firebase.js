// NOTE: import only the Firebase modules that you need in your app... except
// for the second line, which makes both the linter and react-firebase happy
import firebase from 'firebase/compat/app';
import 'firebase/compat/firestore';

// Initalize Firebase.
const config = {
    apiKey: "AIzaSyDvVB6SIAhXqa3chcJgaJi_XGkd1W3xIhI",
    authDomain: "tcl-1-smart-shopping-list.firebaseapp.com",
    projectId: "tcl-1-smart-shopping-list",
    storageBucket: "tcl-1-smart-shopping-list.appspot.com",
    messagingSenderId: "490626563800",
    appId: "1:490626563800:web:1c6c138d68d09a6f8de1ad"
};

let fb = firebase.initializeApp(config);

export { fb };
