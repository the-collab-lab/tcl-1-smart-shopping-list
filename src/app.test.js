import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { fb as firebase } from './lib/firebase.js';
import { FirestoreProvider } from 'react-firestore';
import App from './app';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <FirestoreProvider firebase={firebase}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </FirestoreProvider>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot on load', () => {
  const tree = TestRenderer.create(
    <FirestoreProvider firebase={firebase}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </FirestoreProvider>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
