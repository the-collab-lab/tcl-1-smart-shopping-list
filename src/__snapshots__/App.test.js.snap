// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot on load 1`] = `
Array [
  <main
    className="page-wrapper css-vdsylo"
  >
    <header
      className="header css-527nkd"
    >
      <a
        className="back-link css-nuywz6"
        href="/"
        onClick={[Function]}
        visualstate="hidden"
      >
        Go Back
      </a>
      <a
        className="title-link css-gq9b4g"
        href="/"
        onClick={[Function]}
        visualstate="default"
      >
        <h1
          className="css-r73csp"
        >
          iNeedToBuy
        </h1>
      </a>
      <a
        className="example-link css-gq9b4g"
        href="http://app.ineedtobuy.xyz/"
        target="_blank"
        visualstate="default"
      >
        See Example
      </a>
    </header>
    <section
      className="content-wrapper css-1xpf03y"
    >
      <p
        className="loading"
      >
        Loading...
      </p>
    </section>
    <footer
      className="footer css-1i3mvty"
    >
      <a
        className="list-link css-gq9b4g"
        href="/"
        onClick={[Function]}
        visualstate="default"
      >
        My List
      </a>
      <a
        className="add-item-link css-gq9b4g"
        href="/add-item"
        onClick={[Function]}
        visualstate="default"
      >
        Add Item
      </a>
    </footer>
  </main>,
  " ",
]
`;
